cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java

```diff
@@ -130,12 +130,22 @@ public void add(List<Fact> facts) {
   }
 
   public FactScanner scan(FactScan scan) {
-    return new FactScanner(getScanner(scan), codec, scan.getStartTs(), scan.getEndTs());
+    return new FactScanner(getScanner(scan), codec, scan.getStartTs(), scan.getEndTs(), scan.getMeasureNames());
   }
 
   private Scanner getScanner(FactScan scan) {
-    byte[] startRow = codec.createStartRowKey(scan.getTagValues(), scan.getMeasureName(), scan.getStartTs(), false);
-    byte[] endRow = codec.createEndRowKey(scan.getTagValues(), scan.getMeasureName(), scan.getEndTs(), false);
+    String firstMeasureName = null;
+    String lastMeasureName = null;
+
+    if (scan.getMeasureNames() != null && !scan.getMeasureNames().isEmpty()) {
+      List<String> measures = Lists.newArrayList(scan.getMeasureNames());
+      Collections.sort(measures);
```

Comment:
if null , we scan all the metrics in the context right, wouldn't that be expensive ? 
should we convert to byte arrays and then sort lexicographically to determine start and end rows ?